### PR TITLE
ReverseSequence op

### PR DESCRIPTION
**Pull request by CharlieL7**
Implements the ReverseSequence ONNX operator as a parser.

* This parser can only handle a constant sequence_lens input. This is the same as what is handled for TensorRT as far as I can tell.
* We could handle a variable sequence_lens input; that would require ref and GPU implementations of the operator.
* The ONNX backend tests are disabled because this does not handle variable sequence_lens.

### CODE REVIEW

File: src/onnx/parse_reversesequence.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        }
+
+        instruction_ref ret;
+
+        for(int b = 0; b < batch_size; ++b)
+        {
+            instruction_ref s0;
+            if(sequence_lens[b] > 1)
+            {
+                s0 = info.add_instruction(make_op("slice",
```
Comment (kahmed10) [inline src/onnx/parse_reversesequence.cpp:89]: seems like most of the fields for the operator are the same with slight variations. Maybe you can make a helper function `add_slice()` where the starts and ends are modified for each call?

Comment (kahmed10) [inline src/onnx/parse_reversesequence.cpp:89]: seems like most of the fields for the operator are the same with slight variations. Maybe you can make a helper function `add_slice()` where the starts and ends are modified for each call?

File: test/onnx/onnx_test.cpp [hunk 2]
```diff
@@ -4173,6 +4173,133 @@ TEST_CASE(resize_upsample_pf_test)
...
+    int time_size  = sx.lens()[time_axis];
+
+    auto ret = mm->add_instruction(
+        migraphx::make_op(
+            "slice",
+            {{"axes", {batch_axis, time_axis}}, {"starts", {0, 0}}, {"ends", {1, time_size}}}),
+        input);
+    for(int b = 1; b < batch_size; ++b)
+    {
+        auto s0 = mm->add_instruction(migraphx::make_op("slice",
```
Comment (kahmed10) [inline test/onnx/onnx_test.cpp:4200]: same feedback as I mentioned in the parser.

Comment (kahmed10) [inline test/onnx/onnx_test.cpp:4200]: same feedback as I mentioned in the parser.

**Review by kahmed10 — APPROVED** [review]

**Review by turneram — APPROVED** [review]

File: src/onnx/parse_reversesequence.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            time_axis = info.attributes.at("time_axis").i();
+        }
+        if(time_axis != 0 and time_axis != 1)
+        {
+            MIGRAPHX_THROW("REVERSESEQUENCE: time axis not 0 or 1");
+        }
+
+        if(time_axis == batch_axis)
+        {
+            MIGRAPHX_THROW("REVERSESEQUENCE: time axis and batch axis are the same");
```
Comment (turneram) [inline src/onnx/parse_reversesequence.cpp:50]: Any clue why these are still being reported as not covered?

Comment (CharlieL7) [reply src/onnx/parse_reversesequence.cpp:50]: Figured it out, there was a typo in the gen_onnx.py.

**Pull request merged**